packages/split-button/src/Menu/Menu.tsx

```diff
@@ -0,0 +1,162 @@
+  const buttonRef = useRef<null | HTMLButtonElement>(null);
+  const menuRef = useRef<null | HTMLDivElement>(null);
+
+  const setOpen =
+    (typeof controlledOpen === 'boolean' && controlledSetOpen) ||
+    uncontrolledSetOpen;
+  const open = controlledOpen ?? uncontrolledOpen;
+
+  const handleTriggerClick: MouseEventHandler = e => {
+    onTriggerClick?.(e);
+    if (typeof controlledOpen !== 'boolean') {
+      setOpen(o => !o);
+    }
+  };
+
+  const handleClose = () => {
+    buttonRef.current?.focus();
+    setOpen(false);
+  };
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    switch (e.keyCode) {
+      case keyMap.Tab:
+      case keyMap.Escape:
+        handleClose();
+        break;
+    }
+  };
+
+  useEventListener('keydown', handleKeyDown, { enabled: open });
+  useBackdropClick(handleClose, [buttonRef, menuRef], open);
+
+  const renderMenuItems = () => {
+    const onChildClick = (
+      e: MouseEvent,
+      menuItem: ReactElement<MenuItemProps>,
+    ) => {
+      handleClose();
+      // @ts-expect-error - onClick is inferred from inferredPolymorphic
+      menuItem.props.onClick?.(e);
+    };
+
+    const renderMenuItem = (
+      menuItem: ReactElement<MenuItemProps>,
+      index = 0,
+    ) => {
+      if (isComponentType(menuItem, 'MenuItem')) {
+        return React.cloneElement(menuItem, {
+          active: false,
+          key: `menuItem-${index}`,
+          // @ts-expect-error - onClick is inferred from inferredPolymorphic
+          onClick: (e: MouseEvent) => onChildClick(e, menuItem),
+        });
+      } else {
+        console.warn('Please use a LeafyGreen <MenuItem> component.');
+      }
+    };
+
+    if (menuItems) {
+      if ('props' in menuItems) {
+        if (!isEmpty(menuItems.props)) {
+          // Array of menuItems
+          if (Array.isArray(menuItems.props.children)) {
+            return menuItems.props.children.map(
+              (
+                menuItem: ReactElement<MenuItemProps>,
+                index: number,
+              ): ReactElement<MenuItemProps> | undefined => {
+                if (menuItem == null) {
+                  return menuItem;
+                }
+
+                return renderMenuItem(menuItem, index);
+              },
+            );
+          }
+
+          // Object with one menuItem
+          // React returns an object instead of an array if there is one item
+          return renderMenuItem(menuItems.props.children);
+        }
+      }
+    }
+  };
+
+  return (
+    <>
+      <Button
+        type="button"
+        disabled={disabled}
+        leftGlyph={<Icon glyph="CaretDown" />}
+        variant={variant}
+        size={size}
+        baseFontSize={baseFontSize}
+        className={cx(triggerBaseStyles, triggerSizeStyles[size!], {
+          [triggerThemeStyles(theme, variant!)]: !disabled,
+        })}
+        aria-label="More options" // TODO: should this be a prop for consumers?
```

Comment:
Yes, this should be configurable by the consumer. "More options" is a fine default, but there will probably be a better contextual label